libraries/tock-register-interface/README.md

```diff
@@ -3,130 +3,115 @@
 This crate provides an interface and types for defining and
 manipulating registers and bitfields.
 
-## Defining registers
-
-The crate provides three types for working with memory mapped registers:
-`ReadWrite`, `ReadOnly`, and `WriteOnly`, providing read-write, read-only, and
-write-only functionality, respectively. These types implement the `Readable`,
-`Writeable` and `ReadWriteable` traits.
-
-Defining the registers is done with the `register_structs` macro, which expects
-for each register an offset, a field name, and a type. Registers must be
-declared in increasing order of offsets and contiguously. Gaps when defining the
-registers must be explicitly annotated with an offset and gap identifier (by
-convention using a field named `_reservedN`), but without a type. The macro will
-then automatically take care of calculating the gap size and inserting a
-suitable filler struct. The end of the struct is marked with its size and the
-`@END` keyword, effectively pointing to the offset immediately past the list of
-registers.
+## Defining peripherals
 
-```rust
-use tock_registers::registers::{ReadOnly, ReadWrite, WriteOnly};
+An MMIO peripheral is defined using the `peripheral!` macro:
 
-register_structs! {
+```rust
+tock_registers::peripheral! {
     Registers {
```

Comment:
rustdoc is supported in this position as well, correct? Consider adding a `/// Documentation for Registers` or similar line here

Reply:
Yes, it is. I added lines like that in both README and `peripheral!` to make it clear that doc comments are supported on both peripheral definitions and fields.